src/methods/join.js

```diff
@@ -151,6 +151,11 @@ export class JoinImplementation {
     pnc: PossiblyNormalCompletion,
     c: PossiblyNormalCompletion
   ): PossiblyNormalCompletion {
+    //merge the two pathConditions
+    let mergedPath = [];
+    mergedPath = mergedPath.concat(pnc.pathCondtion);
+    mergedPath = mergedPath.concat(c.pathCondtion);
```

Comment:
You can shorten this to:

let mergedPath = pnc.pathConditions.concat(c.pathConditions);